bittensor/synapse.py

```diff
@@ -513,8 +513,7 @@ def body_hash(self) -> str:
         hashes = []
 
         # Getting the fields of the instance
-        instance_fields = self.__dict__
-
+        instance_fields = self.dict() # __dict__ converts the Tensor types into strings causing a mismatch in the hash, so we use .dict() instead
```

Comment:
I think @ifrit98 would have better visibility on this change than I do, so we should for his approval before merging.

Reply:
@ifrit98 would this create an issue for the header size? If I am not mistaken, there is a sizing limit on the headers which caused us to create hashes rather than using the full tensor.